noria-server/src/controller/migrate/mod.rs

```diff
@@ -205,47 +205,204 @@ impl<'a> Migration<'a> {
         self.mainline.graph()
     }
 
-    fn ensure_reader_for(&mut self, n: NodeIndex, name: Option<String>) {
+    fn ensure_reader_for(&mut self, n: NodeIndex, name: Option<String>, num_replicas: usize) {
         if !self.readers.contains_key(&n) {
-            // make a reader
-            let r = node::special::Reader::new(n);
-            let r = if let Some(name) = name {
-                self.mainline.ingredients[n].named_mirror(r, name)
-            } else {
-                self.mainline.ingredients[n].mirror(r)
-            };
-            let r = self.mainline.ingredients.add_node(r);
-            self.mainline.ingredients.add_edge(n, r, ());
-            self.readers.insert(n, r);
+            let mut readers = vec![];
+            for i in 0..num_replicas {
+                let replica_i = {
+                    if num_replicas == 0 {
+                        None
+                    } else {
+                        Some(i)
+                    }
```

Comment:
Is this distinction really important? Do we need to distinguish between a view with one materialization and with a replication factor of 1 (which would also just have one materialization)?